OpenRA.Mods.Common/Orders/PlaceBuildingOrderGenerator.cs

```diff
@@ -251,7 +252,7 @@ IEnumerable<IRenderable> IOrderGenerator.RenderAboveShroud(WorldRenderer wr, Wor
 
 				footprint.Add(topLeft, MakeCellType(AcceptsPlug(topLeft, plugInfo)));
 			}
-			else if (lineBuildInfo != null)
+			else if (owner.Shroud.IsExplored(topLeft) && lineBuildInfo != null)
```

Comment:
Can we please swap the two conditions? `lineBuildInfo != null && owner.Shroud.IsExplored(topLeft)`
`IsExplored` is relatively expensive compared to a simple null check, so it makes sense to check that first and not do the call to `IsExplored` if it is not needed.

Reply:
Sure, I'll do that.